### PR TITLE
Jetpack connect plans page

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR creates a new plans page to redirect the users after they connect their jetpack site via jetpack-connect. 

image

untitled screencast 12

Also, this fix https://github.com/Automattic/wp-calypso/issues/5149 (as a side effect)

How to test
========
1. You need a .org site running the latest master version of jetpack
2. Disconnect your Jetpack from .com, if needed
3. Go to http://calypso.localhost:3000/jetpack/connect and enter your site
4. After a while, you should be taken to that new plans page

ping @roccotripaldi @rickybanister and someone who has worked with plans before... @drewblaisdell ? @stephanethomas ?

### CODE REVIEW

**Comment by roccotripaldi** [comment]
Niice. Looking pretty good. I did the rebase for you!
I noticed a few things:

The Jetpack logo is a bit high and it's getting eaten by the header.
Also, the site I completed the process with is already on a Premium plan, but the UI did not reflect that:
screen shot 2016-05-17 at 3 17 48 pm

**Comment by rickybanister** [comment]
If you already have a plan of any kind we need to be skipping this step. That would be the case if it's a re-connection I suppose.

**Comment by johnHackworth** [comment]
@rickybanister ok, now it skips the plan page when the site already has a business or a premium plan

@roccotripaldi  I removed the jetpack logo (that was part of the new layout, but I have forgotten to remove it, thank you :D ), and now it looks like this:

image

**Comment by johnHackworth** [comment]
fixes https://github.com/Automattic/wp-calypso/issues/4733

**Comment by roccotripaldi** [comment]
LGTM! 🚢

**Comment by beaulebens** [comment]
I haven't been able to get this to work/show up. Am I missing something?

- Calypso running this branch
- Jetpack site running `master` (disconnected, active)
- Just in case, cleared IndexedDB for Calypso
- Go to http://calypso.localhost:3000/jetpack/connect
- Enter URL for Jetpack site, hit enter
- Some things happen, then I land on http://calypso.localhost:3000/jetpack/connect/authorize
- Click the Approve button
- End up back in wp-admin with my site connected (never saw a plans page).

**Comment by beaulebens** [comment]
Blarg! This test site has a Premium plan already, so I guess it's silently skipping?

**Comment by johnHackworth** [comment]
No, even if you have a premium plan & the plans page is being skipped, you shouldn't end in your wp-admin if you started in calypso. In fact, you shouldn't see the 'approve' button, it should get auto approved. 

Long shot: I use to get to this (erroneous) flow from time to time when I start it in calypso.dev:3000 instead of calypso.localhost:3000 (yeah, all these months and I still use .dev), since you are redirected back to .localhost and any info persisted by .dev would not be available. Could that be the case here?

**Comment by johnHackworth** [comment]
hmmm, I'm not sure what has happened, but I was getting the same result than you in my localhost...

I've rebased with master and that fixed the issue

/shrug

If you update the branch, it should be working now

**Comment by johnHackworth** [comment]
@beaulebens Since design & codewise seems to be ok, and the rebase seems to have fixed the bug you had, I'm going to merge this so we can test everything together in master

File: client/state/jetpack-connect/reducer.js [hunk 1]
```diff
@@ -52,17 +52,30 @@ export function jetpackConnectSessions( state = {}, action ) {
 	switch ( action.type ) {
 		case JETPACK_CONNECT_STORE_SESSION:
 			return Object.assign( {}, state, buildNoProtocolUrlObj( action.url ) );
-		case SERIALIZE:
+		case JETPACK_CONNECT_SSO_AUTHORIZE_SUCCESS:
```
Comment (ebinnion) [inline client/state/jetpack-connect/reducer.js]: I removed this in #5406. Perhaps, a bad rebase?

Comment (johnHackworth) [inline client/state/jetpack-connect/reducer.js]: damn.. between yesterday & today, I merged 9 different branches modifying this files and it has been a nightmare of conflicts and weird rebases ... sorry, I'll remove it

**Pull request merged**